src/Tools/dotnet-dump/ExtensionCommands/TaskStateCommand.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.Diagnostics.Repl;
+
+namespace Microsoft.Diagnostic.Tools.Dump.ExtensionCommands
+{
+    [Command(Name = "taskstate", AliasExpansion = "TaskState", Help = "Display a Task state in a human readable format.")]
+    [CommandAlias(Name = "tks")]
+    public class TaskStateCommand : ExtensionCommandBase
+    {
+        [Argument(Help = "The Task instance address.")]
+        public string Address { get; set; }
+
+        [Option(Name = "--value", Help = "<value> is the value of a Task m_stateFlags field.")]
+        [OptionAlias(Name = "-v")]
+        public ulong? Value { get; set; }
+
+
+        public override void Invoke()
+        {
+            if (string.IsNullOrEmpty(Address) && !Value.HasValue)
+            {
+                WriteLine("Missing Task reference address or state value..." + Environment.NewLine);
+                return;
+            }
+
+            ulong stateFlag = Value.GetValueOrDefault();
+
+            // access the Task state field if the flag is not given as a parameter
+            if (!Value.HasValue)
+            {
+                // skip 0x or leading 0000 if needed 
+                if (Address.StartsWith("0x"))
+                    Address = Address.Substring(2);
+                Address = Address.TrimStart('0');
+                
+                if (!ulong.TryParse(Address, System.Globalization.NumberStyles.HexNumber, System.Globalization.CultureInfo.InvariantCulture, out var address))
+                {
+                    WriteLine("Numeric value expected: either a task address or -v <state value>..." + Environment.NewLine);
+                    return;
+                }
+
+                // check if it is a task address
+                stateFlag = Helper.GetTaskStateFromAddress(address);
+                if (stateFlag == 0)
+                {
+                    // otherwise, it might be a valid flag but in decimal format
+                    if (!ulong.TryParse(Address, System.Globalization.NumberStyles.Integer, System.Globalization.CultureInfo.InvariantCulture, out stateFlag))
```

Comment:
You should always use the state flag option and not over load Address if possible.

Reply:
Right: this is less confusing

Reply:
I have pushed a commit with the required changes (except for hexa parsing that I will change when the global parsing is fixed)